src/features/vote/useTableValues.ts

```diff
@@ -0,0 +1,137 @@
+import { useState, useEffect } from "react";
+import { EncryptedVote } from "web3/get/queryEncryptedVotesEvents";
+import { PendingRequest } from "web3/get/queryGetPendingRequests";
+import { DateTime } from "luxon";
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
+import { ethers } from "ethers";
+import { PostRevealData } from "web3/post/revealVotes";
+import web3 from "web3";
+
+interface TableValue {
+  ancillaryData: string;
+  identifier: string;
+  vote: string;
+  revealed: boolean;
+  ancHex: string;
+  timestamp: string;
+}
+
+export default function useTableValues(
+  activeRequests: PendingRequest[],
+  encryptedVotes: EncryptedVote[],
+  revealedVotes: VoteRevealed[]
+) {
+  const [tableValues, setTableValues] = useState<TableValue[]>([]);
+  const [postRevealData, setPostRevealData] = useState<PostRevealData[]>([]);
+
+  // Take activeRequests and encryptedVotes and convert them into tableViews
+  useEffect(() => {
+    // Check if the user has voted in this round.
+    if (activeRequests.length && !encryptedVotes.length) {
+      const tv: TableValue[] = activeRequests.map((el) => {
+        return {
+          ancillaryData: el.ancillaryData,
+          vote: "-",
+          identifier: el.identifier,
+          revealed: false,
+          ancHex: el.idenHex,
+          timestamp: DateTime.fromSeconds(Number(el.time)).toLocaleString({
+            month: "short",
+            day: "2-digit",
+            year: "numeric",
+            hour: "2-digit",
+            minute: "2-digit",
+            hourCycle: "h24",
+            timeZoneName: "short",
+          }),
+        };
+      });
+
+      setTableValues(tv);
+    }
+    if (activeRequests.length && encryptedVotes.length) {
+      const tv = [] as TableValue[];
+      const postData = [] as PostRevealData[];
+
+      activeRequests.forEach((el) => {
+        const datum = {} as TableValue;
+        datum.ancillaryData = el.ancillaryData;
+        datum.identifier = el.identifier;
+        let vote = "-";
+        // I believe latest events are on bottom. requires testing.
+        const latestVotesFirst = [...encryptedVotes].reverse();
```

Comment:
this keeps moving back into the loop. dont think it needs to keep being re-created. if you are planning on refactoring this logic though dont worry about it in this PR